components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPTransportSender.java

```diff
@@ -168,29 +180,24 @@ public void sendMessage(MessageContext messageContext, String targetEPR,
                     OMElement payLoad,body;
                     body = messageContext.getEnvelope().getBody();
                     payLoad = body.getFirstChildWithName(new QName(RFCConstants.BAPIRFC));
-                    if (log.isDebugEnabled()){
-                        log.debug("Received RFC/Meta DATA: " + payLoad);
-                    }
-                    String rfcFunctionName = RFCMetaDataParser.getBAPIRFCFucntionName(payLoad);
+                    log.info("Received RFC/Meta DATA: " + payLoad);
+                    String rfcFunctionName = RFCMetaDataParser.getBAPIRFCFunctionName(payLoad);
+                    log.info("RFC name: " + rfcFunctionName);
                     if (isTransaction(messageContext) || isLogon) {
-                        log.info("Begin Transaction");
+                        if (log.isDebugEnabled()) {
+                            log.debug("Beginning Transaction");
```

Comment:
If this is called in parallel, we need to identify "Beginning Transaction" on which connection?

Reply:
Same goes with every log statement.

Reply:
We cannot access connection detaild. Logged the function that was being called at the time.